File: gym_management_task/gym_management_task/report/journey/journey.js
frappe.query_reports["Journey"] = {
	"filters": [
		{
			"fieldname": "member",
			"label": __("Member"),
			"fieldtype": "Link",
			"options": "Gym Member",
			"mandatory": 1,
			"default": frappe.session.user
		}
	],
	onload: function (report) {
		console.log("Report is being loaded");
		if (!frappe.query_report.chart) {
			frappe.query_report.chart = new Chart({
				parent: report.page.main,
				title: "Journey",
				type: 'bar', // or the chart type you prefer
				height: 250,
				data: {}
			});
		}
		frappe.query_report.get_chart_data(report);
	},
	// Chart Data Configuration
	get_chart_data: function (columns, result) {
		console.log(result);  // Add this to log the result for debugging
		let labels = [];
		let calorie_intake = [];
		let weight_loss = [];
		let weight_gain = [];
		let body_mass = [];

		result.forEach(row => {
			labels.push(row.date);  // Assuming date is the first column
			calorie_intake.push(row.calorie_intake);  // Assuming Calorie Intake is second column
			weight_loss.push(row.weight_loss);  // Assuming Weight Loss is third column
			weight_gain.push(row.weight_gain);  // Assuming Weight Gain is fourth column
			body_mass.push(row.weight_loss);  // Assuming Body Mass is fifth column
		});
		console.log(labels);
		console.log(calorie_intake);
		console.log(weight_loss);
		console.log(weight_gain);
		console.log(body_mass);
		return {
			data: {
				labels: labels,
				datasets: [
					{ name: "Calorie Intake", values: calorie_intake },
					{ name: "Weight Loss", values: weight_loss },
					{ name: "Weight Gain", values: weight_gain },
					{ name: "Body Mass", values: body_mass }
				]
			},
			type: 'bar',
			height: 250
		};
	}
};

File: gym_management_task/gym_management_task/report/journey/journey.py
# Copyright (c) 2024, ateeq and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
    if not filters:
        filters = {}

    # Check if the member filter is selected; if not, prompt the user
    if not filters.get("member"):
        frappe.throw("Please select a member to view their fitness journey.")

    # Fetch data from the child table (Tracking Sheet) based on the selected Gym Member
    data = frappe.db.sql("""
        SELECT 
            ts.date AS "Date", 
            ts.calorie_intake AS "Calorie Intake", 
            ts.weight_loss AS "Weight Loss", 
            ts.weight_gain AS "Weight Gain", 
            ts.body_mass AS "Body Mass"
        FROM `tabTracking Sheet` ts
        JOIN `tabProcess Sheet` ps ON ps.name = ts.parent
        WHERE ps.member = %(member)s
        ORDER BY ts.date ASC
    """, {"member": filters.get("member")}, as_dict=1)

    # Debugging: Print the fetched data for confirmation
    frappe.msgprint(f"Fetched data: {data}")

    # Define columns for the report table
    columns = [
        {"label": "Date", "fieldname": "Date", "fieldtype": "Date", "width": 120},
        {"label": "Calorie Intake", "fieldname": "Calorie Intake", "fieldtype": "Float", "width": 120},
        {"label": "Weight Loss", "fieldname": "Weight Loss", "fieldtype": "Float", "width": 120},
        {"label": "Weight Gain", "fieldname": "Weight Gain", "fieldtype": "Float", "width": 120},
        {"label": "Body Mass", "fieldname": "Body Mass", "fieldtype": "Float", "width": 120}
    ]

    # Return columns and fetched data
    return columns, data

